### PR TITLE
Move external host tags payload to agentchecks provider

### PR DESCRIPTION
**Pull request by masci**
### What does this PR do?

Send external host tags payload along with the `agent_checks` provider instead of the `host` provider.

### Motivation

The `host` provider sends the payload every 4 hours, too much for external host tags. `agent_checks` provider sends every 10 minutes which is fine and closer to what we did in A5.

### Additional Notes

~Will run the artifact from this branch to confirm it's working as expected~
Check works as expected with an agent built from this branch.

### CODE REVIEW

File: pkg/collector/metadata/agentchecks/agentchecks.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/collector/autodiscovery"
 	"github.com/DataDog/datadog-agent/pkg/collector/runner"
 	"github.com/DataDog/datadog-agent/pkg/metadata/common"
+	"github.com/DataDog/datadog-agent/pkg/metadata/externalhost"
```
Comment (truthbk) [inline pkg/collector/metadata/agentchecks/agentchecks.go]: I noticed the implementation for the externalhost cache doesn't look thread-safe.

**Review by truthbk — COMMENTED** [review]
This looks good to me, but I think the `externalhost` cache might not be thread-safe, probably worth looking into - I think we could have some potential race condition there.

File: pkg/collector/metadata/agentchecks/agentchecks.go (hunk 1 above)
Comment (masci) [reply pkg/collector/metadata/agentchecks/agentchecks.go]: it's not indeed, good catch, will address in this PR since it shouldn't be much code

**Review by truthbk — APPROVED** [review]
I was going to suggest an `RWLock` instead, but actually we'll typically have the opposite case, multiple writers and a single reader. So we'd probably get little gains anyway, also it's not going to be super intensive activity so 👍  :shipit:

**Pull request merged**